### DISCUSSION TITLE
[Feature]: Fastly Compute@Edge Adapter

### DISCUSSION THREAD

**Discussion by s-mark-holmes-ii** [remix-run/remix#2888] [Proposals]
### What is the new or updated feature that you are suggesting?

I would love to have an adapter built for Fastly Compute@Edge. Here is the relevant documentation:

1. https://developer.fastly.com/learning/compute/javascript/
2. https://developer.fastly.com/learning/compute/testing/

Seems to be relatively close to how one would deploy to Cloudflare workers; however, I am a bit lost on the proper way of building an adapter. I would be more than happy to assist, but I am unaware of what your priorities/product timelines are for alternative adapter support.

### Why should this feature be included?

Fastly seems to be a fairly popular CDN (https://www.fastly.com/customers/) that now supports deploying apps to the edge. We are beginning a major platform redesign at a major digital music streaming service. My case for using Remix would be strongly supported if I could show them that we could deploy to Fastly, which is unfortunately a hard requirement. Thanks for your time and consideration!

**Comment by s-mark-holmes-ii** [comment]
@kailan For visibility.

**Comment by s-mark-holmes-ii** [comment]
As an update to this thread @miekapage has successfully gotten remix running on Fastly Compute@Edge here: https://web-compute-test.edgecompute.app 🎉 . @miekapage If you have any questions/concerns/stumbling blocks that you run into, feel free to post here for posterity. Thanks!

**Comment by quinnlangille** [comment]
+1 for this, would be interested in seeing how compute@edge runs. Interesting to my org both for long term use and migrating parts of our currently fastly set up -> other edge.

If there's any learnings/quirks from what you guys are testing out would love to hear! Also, if Remix is open to the contribution then would definitely land a hand if needed

**Comment by miekapage** [comment]
just a note about this -- even if a new adaptor wasn't made for compute@edge specifically, it would be nice for remix to provide a platform-agnostic postinstall script option that doesn't force installing either node or cloudflare dependencies. thanks !

**Comment by bjrn** [comment]
@miekapage, was just about to look into this – would you be open to sharing your code in some way for collaboration?

**Comment by miekapage** [comment]
hey @bjrn absolutely ! just invited you to a repo https://github.com/miekapage/remix-with-fastly

**Comment by bjrn** [comment]
Thanks @miekapage! Will take a look during this week 👍🏼

**Comment by marfillaster** [comment]
@miekapage interested in subscribing in https://github.com/miekapage/remix-with-fastly as well.

**Comment by jacob-ebey** [comment]
Can we avoid posting links to private repos? If it's something you want to share here please make it public so everyone can access it.

**Comment by miekapage** [comment]
> Can we avoid posting links to private repos? If it's something you want to share here please make it public so everyone can access it.

[remix-with-fastly](https://github.com/miekapage/remix-with-fastly) is now public @jacob-ebey @marfillaster @Girish21

@bjrn I removed git history and force pushed before making it public, so apologies if you've got anything local.

**Comment by JakeChampion** [comment]
Fastly have also built a Remix adapter and are aiming to get it added into Remix -> https://github.com/remix-run/remix/discussions/4613 🤞

**Comment by harmony7** [comment]
Have created this PR
https://github.com/remix-run/remix/pull/4860

Once this is accepted, it will be possible to use the adapter we created at
https://github.com/fastly/remix-compute-js

**Reply by MichaelDeBoey** [reply]
#4841 is now merged, which will provide new extra config options (`serverConditions`, `serverMainFields` & `serverMinify`) that can be used
